src/content/en/2022/javascript.md

```diff
@@ -16,7 +16,864 @@ featured_stat_2: TODO
+    sheets_gid="397179470",
+    sql_file="resource-hints-preload-prefetch-modulepreload-distribution.sql"
+  )
+}}
+
+Adoption of `prefetch` here is somewhat surprising, with three `prefetch` hints for JavaScript resources per page. However, the number of these hints at the 75th and 90th percentiles suggests that there may be a fair amount of waste in the form of unused resources for page navigations that never occur.
+
+{{ figure_markup(
+    image="preload.png",
+    caption="Distribution of `preload` adoption for JavaScript resources per page.",
+    description="TODO",
+    chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTok8UGqYaA58uKaASB2pfM2jlmQu6g3kwHxB6Lb8L7dbccyDtQht823YhchdXRG8SZpB3asRayJI97/pubchart?oid=1371197443&format=interactive",
+    sheets_gid="397179470",
+    sql_file="resource-hints-preload-prefetch-modulepreload-distribution.sql"
+  )
+}}
+
+Remember—this analysis tracks how many resource hints are used for _JavaScript_ resources on pages that use one or more `preload` hints. The median page is delivering two `preload` hints for JavaScript, which isn't bad on its face, but it often depends on the _size_ of the script, how much processing scripts can kick off, or whether the script fetched via `preload` is even needed for the initial page load.
+
+Unfortunately, we see five `preload` hints for JavaScript resources at the 90th percentile, which may be too much. This suggests that pages at the 90th percentile are especially reliant on JavaScript, and are using preload to try and overcome the performance issues that result.
+
+{{ figure_markup(
+    image="modulepreload.png",
+    caption="Distribution of `modulepreload` adoption for JavaScript resources per page.",
+    description="TODO",
+    chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTok8UGqYaA58uKaASB2pfM2jlmQu6g3kwHxB6Lb8L7dbccyDtQht823YhchdXRG8SZpB3asRayJI97/pubchart?oid=100852946&format=interactive",
+    sheets_gid="397179470",
+    sql_file="resource-hints-preload-prefetch-modulepreload-distribution.sql"
+  )
+}}
+
+With `modulepreload`, we see a staggering 6 hints at the 75th percentile, and _14_ at the 90th percentile. This suggests that, while pages using one or more `modulepreload` hints at upper percentiles are shipping untransformed ES6 modules directly to the browser, the need for so many resource hints suggests an overreliance on JavaScript at the upper range.
+
+Resource hints are great tools for optimizing how we load resources in the browser, but if there's one piece of advice you can heed when using them, it's to use them sparingly, and for resources that may not be initially discoverable; for example, a JavaScript file initially loaded in the DOM that requests another one. Rather than preloading loads of scripts, try to whittle down the _amount_ of JavaScript you're shipping, as that will lead to a better user experience rather than preloading gobs of it.
```

Comment:
```suggestion
Resource hints are great tools for optimizing how we load resources in the browser, but if there's one piece of advice you can heed when using them, it's to use them sparingly, and for resources that may not be initially discoverable; for example, a JavaScript file initially loaded in the DOM that requests another one. Rather than preloading loads of scripts, try to whittle down the amount of JavaScript you're shipping, as that will lead to a better user experience rather than preloading gobs of it.
```

Also is "gobs" a good word here?

Reply:
Reopening cause of my "gobs" comment. Happy to go with it if you are, but think it's a bit of a weird word and non-English natives may not get it?

Reply:
Maybe. I'm fairly familiar with a fair bit of slang from the region, but maybe s/gobs/tons?